pkg/controllers/mcs/service_import_controller.go

```diff
@@ -0,0 +1,133 @@
+package mcs
+
+import (
+	"context"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog/v2"
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	mcsv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+// ServiceImportControllerName is the controller name that will be used when reporting events.
+const ServiceImportControllerName = "service-import-controller"
+
+// ServiceImportController is to sync derived service from ServiceImport.
+type ServiceImportController struct {
+	client.Client
+	EventRecorder record.EventRecorder
+}
+
+// Reconcile performs a full reconciliation for the object referred to by the Request.
+func (c *ServiceImportController) Reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
+	klog.V(4).Infof("Reconciling ServiceImport %s.", req.NamespacedName.String())
+
+	svcImport := &mcsv1alpha1.ServiceImport{}
+	if err := c.Client.Get(context.TODO(), req.NamespacedName, svcImport); err != nil {
+		if errors.IsNotFound(err) {
+			return c.deleteDerivedService(types.NamespacedName{
+				Name:      names.GenerateDerivedServiceName(svcImport.Name),
+				Namespace: svcImport.Namespace,
+			})
```

Comment:
Don't have to delete it since we have set owner reference.

Reply:
I have remove this delete option.